packages/runtime/container-runtime/src/scheduleManager.ts

```diff
@@ -27,63 +61,93 @@ type IRuntimeMessageMetadata = undefined | {
  */
 export class ScheduleManager {
     private readonly deltaScheduler: DeltaScheduler;
-    private batchClientId: string | undefined;
-    private hitError = false;
+    private state: BatchProcessingState = "None";
 
     constructor(
         private readonly deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
         private readonly emitter: EventEmitter,
+        readonly clientId: () => string | undefined,
```

Comment:
>clientId [](http://example.com/codeflow?start=17&length=8)

out of curiosity, can the client id change for the same containerid ?

Reply:
Yes. In case of reconnects, we'll keep the container but then receive a new clientId.